natscontext/context.go

```diff
@@ -0,0 +1,336 @@
+// Copyright 2022-2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package natscontext
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/nats-io/nats.go"
+)
+
+type context struct {
+	Name   string `json:"-"`
+	config *Settings
+	path   string
+}
+
+type nscCreds struct {
+	UserCreds string `json:"user_creds"`
+	Operator  struct {
+		Service []string `json:"service"`
+	} `json:"operator"`
+}
+
+type Settings struct {
+	Name                string `json:"name,omitempty"`
+	Description         string `json:"description"`
+	URL                 string `json:"url"`
+	nscUrl              string
+	SocksProxy          string `json:"socks_proxy"`
+	Token               string `json:"token"`
+	User                string `json:"user"`
+	Password            string `json:"password"`
+	Creds               string `json:"creds"`
+	nscCreds            string
+	NKey                string   `json:"nkey"`
+	Cert                string   `json:"cert"`
+	Key                 string   `json:"key"`
+	CA                  string   `json:"ca"`
+	NSCLookup           string   `json:"nsc"`
+	JSDomain            string   `json:"jetstream_domain"`
+	JSAPIPrefix         string   `json:"jetstream_api_prefix"`
+	JSEventPrefix       string   `json:"jetstream_event_prefix"`
+	InboxPrefix         string   `json:"inbox_prefix"`
+	UserJwt             string   `json:"user_jwt"`
+	ColorScheme         string   `json:"color_scheme"`
+	TLSFirst            bool     `json:"tls_first"`
+	WinCertStoreType    string   `json:"windows_cert_store"`
+	WinCertStoreMatchBy string   `json:"windows_cert_match_by"`
+	WinCertStoreMatch   string   `json:"windows_cert_match"`
+	WinCertStoreCaMatch []string `json:"windows_ca_certs_match"`
+}
+
+// Connect connects to the NATS server configured by the named context, empty name connects to selected context. If the name is a absolute name to a context configuration that will be used
```

Comment:
Probably also worth to document that the `Settings` returned here are the selected context's parameters.

Reply:
done this